daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/TransactionPreprocessor.scala

```diff
@@ -37,6 +37,7 @@ private[preprocessing] final class TransactionPreprocessor(
     node match {
       case _: Node.NodeRollback[_] =>
         // TODO https://github.com/digital-asset/daml/issues/8020
+        // how on earth can we turn a rollback node back into a speedy command?
```

Comment:
it is a bit more tricky than other node: because we have to consider the chilren as well.
The naive idea would be to create a new `Catch` command that take as argument the list of commands obtained by translating the chidren. I will postpone that for later, since it would be used only by canton, and we are not completly clear how we will deal with that exactly.